orderbook/src/database/orders.rs

```diff
@@ -202,7 +202,31 @@ struct OrdersQueryRow {
 }
 
 impl OrdersQueryRow {
+    fn calculate_status(&self) -> OrderStatus {
+        match self.kind {
+            DbOrderKind::Buy => {
+                if !self.sum_buy.is_zero() && self.sum_buy == self.buy_amount {
+                    return OrderStatus::Fulfilled;
+                }
+            }
+            DbOrderKind::Sell => {
+                if !self.sum_sell.is_zero() && self.sum_sell == self.sell_amount {
+                    return OrderStatus::Fulfilled;
+                }
+            }
+        }
+        if self.invalidated {
+            return OrderStatus::Cancelled;
```

Comment:
Yes, that's something I have in mind and I'm not sure.
In one hand, we could flatten all possible combinations, making it a bit verbose.
On the other, partial fill is a orthogonal state that can be combined if the consumer needs it or not.

Reply:
Expanding on that, if we consider partial fills, we'll need 3 additional states:
OpenPartialFill
CancelledPartialFill
ExpiredPartialFill

Or a better name.

Otherwise, keeping as is, the consumer can check whether executed_amount_* > 0

That's what's done on https://github.com/gnosis/gp-ui/blob/develop/src/utils/operator.ts#L30

I have no strong opinion either way.

One could argue that with more states, you'll have to watch for 2 states instead of 1 to consider an order `open`, and so on.